exec/java-exec/src/main/java/org/apache/drill/exec/ops/QueryContext.java

```diff
@@ -283,4 +286,18 @@ public void close() throws Exception {
       closed = true;
     }
   }
+
+  /**
+  * @param stmtType : The type {@link StatementType} e.g. CTAS, ANALYZE of the statement
+  */
+  public void setStatementType(StatementType stmtType) {
```

Comment:
I have changed the code to allow the statement type to be set only once. Please suggest if there is a better approach.